### PR TITLE
Complete calibration choreography

### PR DESCRIPTION
**Pull request by romanroibu**
##### TODO
- [x] Implement persistence for `screen marker`, `single marker`, `natural feature` choreographies, but not `hmd`.
- [x] Remove old calibration plugins from `calibration_routines`
- [ ] Perform basic tests for `hmd` choreography

Low priority
- [ ] Perform extensive tests for all choreographies
- [x] Show choreography plugins in the list of core plugins
- [x] Hide choreography plugins from the Plugin Manager

Waiting for gazer rework
- [ ] Reenable validation

### CODE REVIEW

**Comment by romanroibu** [comment]
@papr the code at `936f63d8ce49ad395aca6a21d95fe8a88fcf70ab` has the screen marker calibration working with the 2d gazer, if you need to test.

File: pupil_src/shared_modules/calibration_choreography/natural_feature_plugin.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class NaturalFeatureChoreographyPlugin(CalibrationChoreographyPlugin):
+    """Calibrate using natural features in a scene.
+        Features are selected by a user by clicking on
+    """
+
+    label = "Manual Calibration Choreography"
```
Comment (papr) [inline pupil_src/shared_modules/calibration_choreography/natural_feature_plugin.py]:
```suggestion
    label = "Natural Feature Calibration Choreography"
```

File: pupil_src/shared_modules/calibration_choreography/single_marker_plugin.py [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+                )
+                draw_polyline(
+                    indicator, color=RGBA(8.0, 0.1, 0.1, 0.8), line_type=gl.GL_POLYGON
+                )
+
+    ### Internal
+
+    def _perform_start(self):
+        # TODO: Is this check duplicated for all calibration choreographies?
+        if not self.g_pool.capture.online:
```
Comment (romanroibu) [inline pupil_src/shared_modules/calibration_choreography/single_marker_plugin.py]: @papr Could you please clarify why this check exists and if it makes sense for all choreography plugins, or only `screen marker` and `single marker`?

File: pupil_src/shared_modules/calibration_choreography/base_plugin.py [hunk 3]
```diff
@@ -163,147 +189,157 @@ def cleanup(self):
...
+    def ref_list(self) -> T.List[dict]:
+        return self.__ref_list
+
+    def on_choreography_started(self, mode: ChoreographyMode):
         self.notify_all(
-            ChoreograthyNotification(
+            ChoreographyNotification(
                 # TODO: Why the subject is always "calibration.started", and not "accuracy_test.started" for accuracy test mode?
-                # mode=self.__current_mode,
+                # mode=mode,
```
Comment (romanroibu) [inline pupil_src/shared_modules/calibration_choreography/base_plugin.py]: @papr Could you please clarify why the previous implementation was sending `calibration.started` notifications, even when the user was triggering an accuracy test? Is it just a copy/paste bug or was it intentional? Also, what parts could depend on the old behavior?

File: pupil_src/shared_modules/calibration_choreography/single_marker_plugin.py (hunk 2 above)
Comment (papr) [reply pupil_src/shared_modules/calibration_choreography/single_marker_plugin.py]: Necessary for all choreos that rely on having a world video, i.e. screen marker, single marker, and natural features.

File: pupil_src/shared_modules/calibration_choreography/base_plugin.py (hunk 3 above)
Comment (papr) [reply pupil_src/shared_modules/calibration_choreography/base_plugin.py]: Can be renamed to `validation.started`. Other instances of `accuracy_test` should be renamed to `validation`.

**Comment by romanroibu** [comment]
Since there is a lot of overlap between this branch and `complete-calibration-choreography`, both branches will be merged into the feature branch `restructured-gaze-calibration-and-mapping` and development will continue there.

**Pull request merged**